spanner/read.go

```diff
@@ -138,10 +142,32 @@ var _ rowIterator = (*RowIterator)(nil)
 // there are no more results. Once Next returns Done, all subsequent calls
 // will return Done.
 func (r *RowIterator) Next() (*Row, error) {
+	mt := r.meterTracerFactory.createBuiltinMetricsTracer(r.ctx)
 	if r.err != nil {
 		return nil, r.err
 	}
-	for len(r.rows) == 0 && r.streamd.next() {
+	// Start new attempt if we don't have one
+	if mt.currOp.currAttempt == nil {
+		mt.currOp.incrementAttemptCount()
+		mt.currOp.currAttempt = &attemptTracer{
+			startTime: time.Now(),
+		}
+	}
+	defer func() {
+		if mt.method != "" {
+			// record the attempt completion
+			if mt.currOp.currAttempt != nil {
+				mt.currOp.currAttempt.setStatus(status.Code(r.err).String())
+				recordAttemptCompletion(&mt)
```

Comment:
This seems to indicate that we can have a situation where the operation completed, but the attempt did not complete (at least not at the same time). What is that situation?